src/nncf/openvino/optimized_functions/functions.py

```diff
@@ -116,7 +116,7 @@ def do_float_quantization(
     :param precomputed_scale: Optional precomputed scale.
     :return: Returns quantized weight tensor and corresponding scale tensor.
     """
-    assert config.mode in [CompressWeightsMode.NF4, CompressWeightsMode.MXFP4, CompressWeightsMode.FP4]
```

Comment:
Using list of supported modes is preferable, as it will not need update on adding new modes.
And it can be defined as variable like `SUPPORTED_MODES`.

Reply:
Introduced variables `OPTIMIZED_COMPRESSION_COMPATIBLE_INT_MODES`, `OPTIMIZED_COMPRESSION_COMPATIBLE_FLOAT_MODES` and their union `OPTIMIZED_COMPRESSION_COMPATIBLE_MODES`. Reused the last one at `weight_lowering.py`.